### ISSUE TITLE
TensorRT tensorrt.py: function convert_savedmodel_to_frozen_graph() doesn't work

### ISSUE THREAD

**Issue by vilmara** [tensorflow/models#5437]
### System information
- **What is the top-level directory of the model you are using**: /models/research/tensorrt/tensorrt.py
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 16.04.5 LTS (GNU/Linux 4.4.0-128-generic x86_64)
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**: v1.10.1-0-g4dcfddc5d1 1.10.1
- **Bazel version (if compiling from source)**: n/a
- **CUDA/cuDNN version**: CUDA Version 9.0.176 / 7.2.1
- **GPU model and memory**: Tesla V100-SXM2-32GB
- **Exact command to reproduce**: `python3 tensorrt.py --savedmodel_dir=/home/resnet_v2_imagenet_savedmodel/1527887769/ --image_file=image.jpg --native --fp32 --fp16 --output_dir=/home/output_tensorrt`

### Describe the problem
I am using the script tensorrt.py with the official saved model downloaded from [http://download.tensorflow.org/models/official/20180601_resnet_v2_imagenet_checkpoint.tar.gz](url), but this command `outputs = [v.name for v in signature_def.outputs.itervalues()` is trhowing this error: 
`outputs = [v.name for v in signature_def.outputs.itervalues()]
AttributeError: 'google.protobuf.pyext._message.MessageMapContainer' object has no attribute 'itervalues'`. 

I made this temporary change to make it work but still needs to be fixed by TensorFlow team: 
`  with tf.Session(graph=graph) as sess:
    tf.saved_model.loader.load(
        sess, meta_graph.meta_info_def.tags, savedmodel_dir)
    frozen_graph_def = tf.graph_util.convert_variables_to_constants(
        sess, graph.as_graph_def(), ["softmax_tensor"]) `

Basically, I omitted the line   `sess, graph.as_graph_def(), output_names) ` and gave the output name manually with  `sess, graph.as_graph_def(), ["softmax_tensor"])` 

### Source code / logs
`Traceback (most recent call last):
  File "tensorrt.py", line 634, in <module>
    main(argv=sys.argv)
  File "tensorrt.py", line 460, in main
    flags.savedmodel_dir, flags.output_dir)
  File "tensorrt.py", line 196, in convert_savedmodel_to_frozen_graph
    outputs = [v.name for v in signature_def.outputs.itervalues()]
AttributeError: 'google.protobuf.pyext._message.MessageMapContainer' object has no attribute 'itervalues'
`

**Comment by vilmara** [comment]
hi @karmel could you help, please?

**Comment by karmel** [comment]
This sounds like a Python 2/3 issue. Care to submit a PR to fix? I think the fix should still allow for the dynamic names, but should better handle the expected type of the proto in py2 and 3.

**Comment by vilmara** [comment]
Hi @benjamintanweihao, I have replaced it as `outputs = [v.name for v in signature_def.outputs.values()]` and working now.  Thanks

@karmel I have submitted the change proposal as recommended by @benjamintanweihao
